src/ppx/command_line_parser.cpp

```diff
@@ -118,25 +133,73 @@ std::optional<CommandLineParser::ParsingError> CommandLineParser::Parse(int argc
         }
         name = name.substr(2);
 
+        std::string_view value = "";
         if (i + 1 < args.size()) {
-            std::string_view nextElem = ppx::string_util::TrimBothEnds(args[i + 1]);
+            auto nextElem = ppx::string_util::TrimBothEnds(args[i + 1]);
             if (!StartsWithDoubleDash(nextElem)) {
-                // We found an option with a parameter.
-                mOpts.AddOption(name, nextElem);
+                // The next element is a parameter for the current option
+                value = nextElem;
                 ++i;
-                continue;
             }
         }
-        // There is no parameter so it's likely a flag
-        if (name.substr(0, 3) == "no-") {
-            mOpts.AddOption(name.substr(3), "0");
+
+        if (auto error = AddOption(name, value)) {
+            return error;
+        }
+    }
+
+    std::vector<std::string> configJsonPaths;
+    configJsonPaths = mOpts.GetOptionValueOrDefault("config-json-path", configJsonPaths);
+    for (const auto& jsonPath : configJsonPaths) {
+        std::ifstream f(jsonPath);
+        if (f.fail()) {
+            return "Invalid --config-json-path: " + jsonPath;
+        }
+        nlohmann::json data = nlohmann::json::parse(f);
+        if (auto error = AddJsonOptions(data)) {
+            return error;
+        }
+    }
+
+    return std::nullopt;
+}
+
+std::optional<CommandLineParser::ParsingError> CommandLineParser::AddJsonOptions(const nlohmann::json& jsonConfig)
+{
+    std::stringstream ss;
+    for (auto it = jsonConfig.cbegin(); it != jsonConfig.cend(); ++it) {
+        ss << it.value();
+        std::string value = ss.str();
+        ss.str("");
+
+        if (value.length() > 0 && value.substr(0, 1) == "[") {
```

Comment:
you probably shouldn't parse the data again. What about using functions like `it->is_array()`.